src/app.jsx

```diff
@@ -37,47 +37,52 @@ import { NavigatorBreadcrumbs } from "./navigatorBreadcrumbs.jsx";
 import { createDirectory, createLink, deleteItem, editPermissions, renameItem } from "./fileActions.jsx";
 import { SidebarPanelDetails } from "./sidebar.jsx";
 import { NavigatorCardHeader } from "./header.jsx";
+import { usePageLocation } from "hooks.js";
 
 const _ = cockpit.gettext;
 
 export const Application = () => {
+    const { options } = usePageLocation();
     const Dialogs = useDialogs();
     const [currentFilter, setCurrentFilter] = useState("");
     const [files, setFiles] = useState([]);
     const [isGrid, setIsGrid] = useState(true);
-    const [path, setPath] = useState(undefined);
     const [sortBy, setSortBy] = useState(localStorage.getItem("cockpit-navigator.sort") || "az");
     const channel = useRef(null);
     const [selected, setSelected] = useState(null);
     const [selectedContext, setSelectedContext] = useState(null);
     const [showHidden, setShowHidden] = useState(false);
-    const [history, setHistory] = useState([]);
-    const [historyIndex, setHistoryIndex] = useState(0);
+    const [historyIndex, setHistoryIndex] = useState({ length: 1, current: 0 });
 
     const onFilterChange = (_, value) => setCurrentFilter(value);
 
+    // eslint-disable-next-line no-restricted-globals
+    useEffect(() => history.replaceState({ index: historyIndex.current }, ""));
+
     useEffect(() => {
         cockpit.user().then(user => {
-            const userPath = user.home.split("/").slice(1);
-            setPath(userPath);
-            setHistory(h => [...h, userPath]);
+            if (options.path === undefined) {
+                cockpit.location.go("?path=root" + user.home);
+            }
         });
-    }, []);
+    }, [options]);
 
+    const path = options.path ? options.path.slice(5).split("/") : [];
     useEffect(() => {
-        if (path === undefined)
+        if (options.path === undefined)
             return;
 
+        const path = options.path ? options.path.slice(5).split("/") : [];
```

Comment:
These hardcoded numbers (5) are wrong.

Reply:
What is wrong about this? I made split it like that to make it match the old path variable